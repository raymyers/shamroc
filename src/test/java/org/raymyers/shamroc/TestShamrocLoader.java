package org.raymyers.shamroc;

import org.approvaltests.Approvals;
import org.junit.jupiter.api.Test;

import java.io.IOException;
import java.io.InputStream;

import static org.junit.jupiter.api.Assertions.assertTrue;

public class TestShamrocLoader {
    @Test
    void basicParse() throws IOException {
        InputStream stream = this.getClass().getResourceAsStream("Shamroc.shamroc");
        ShamrocLoader subject = new ShamrocLoader(stream);
//        System.out.println(subject.getSyntaxErrors());
        assertTrue(subject.valid());
        Approvals.verify(subject.getStringTree());
    }

    @Test
    void transformToAst() throws IOException {
        InputStream stream = this.getClass().getResourceAsStream("Shamroc.shamroc");
        ShamrocLoader subject = new ShamrocLoader(stream);
//        System.out.println(subject.getSyntaxErrors());
        assertTrue(subject.valid());
        Approvals.verify(subject.toAst());
    }
}
