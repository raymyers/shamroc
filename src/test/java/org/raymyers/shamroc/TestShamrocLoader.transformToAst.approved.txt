Program[defs=[LangDef[name=Shamroc, langItems=[SumType[name=Program, sumTypeOptions=[NameOrType[ident1=Ident[string=List], ident2=Optional[Ident[string=Def]]]]], SumType[name=Def, sumTypeOptions=[NameOrType[ident1=Ident[string=DefLang], ident2=Optional.empty]]], Record[name=DefLang, recordFields=[RecordField[nameOrTypes=[NameOrType[ident1=Ident[string=name], ident2=Optional.empty], NameOrType[ident1=Ident[string=String], ident2=Optional.empty]]], RecordField[nameOrTypes=[NameOrType[ident1=Ident[string=List], ident2=Optional[Ident[string=LangItem]]]]]]], SumType[name=LangItem, sumTypeOptions=[NameOrType[ident1=Ident[string=Record], ident2=Optional.empty], NameOrType[ident1=Ident[string=SumType], ident2=Optional.empty]]], SumType[name=Record, sumTypeOptions=[NameOrType[ident1=Ident[string=List], ident2=Optional[Ident[string=RecordField]]]]], Record[name=RecordField, recordFields=[RecordField[nameOrTypes=[NameOrType[ident1=Ident[string=NameOrType], ident2=Optional.empty]]], RecordField[nameOrTypes=[NameOrType[ident1=Ident[string=Optional], ident2=Optional[Ident[string=NameOrType]]]]]]], Record[name=NameOrType, recordFields=[RecordField[nameOrTypes=[NameOrType[ident1=Ident[string=String], ident2=Optional.empty]]], RecordField[nameOrTypes=[NameOrType[ident1=Ident[string=Optional], ident2=Optional[Ident[string=String]]]]]]], Record[name=SumType, recordFields=[RecordField[nameOrTypes=[NameOrType[ident1=Ident[string=String], ident2=Optional.empty]]], RecordField[nameOrTypes=[NameOrType[ident1=Ident[string=List], ident2=Optional[Ident[string=SumTypeOption]]]]]]], SumType[name=SumTypeOption, sumTypeOptions=[NameOrType[ident1=Ident[string=Record], ident2=Optional.empty], NameOrType[ident1=Ident[string=NameOrType], ident2=Optional.empty]]]]]]]